ANMN/temperature_grid_product/anmn_temp_gridded_product.py

```diff
@@ -506,28 +413,27 @@ def cleaning_err_exit():
         shutil.rmtree(fv01_dir)
         sys.exit(1)
 
-def main(incoming_file_path, deployment_code, output_dir, plot_comparaison=False):
-    global s3_bucket_prefix
+def main(incoming_file_path, deployment_code, output_dir, plot_comparison=False):
     global logger
     global fv01_dir
-    s3_bucket_prefix = 'https://s3-ap-southeast-2.amazonaws.com/imos-data'
     logging           = IMOSLogging()
-    logger            = logging.logging_start(os.path.join(os.environ['WIP_DIR'], 'anmn_temp_grid.log'))
-    list_fv01_url     = wfs_request_matching_file_pattern('anmn_ts_timeseries_map', '%%_FV01_%s%%' % deployment_code, s3_bucket_url=True)
+    logger            = logging.logging_start(os.path.join(output_dir, 'anmn_temp_grid.log'))
+    list_fv01_url     = wfs_request_matching_file_pattern('anmn_ts_timeseries_map', '%%_FV01_%s%%' % deployment_code, s3_bucket_url=True, url_column='file_url')
     previous_fv02_url = wfs_request_matching_file_pattern('anmn_all_map', '%%Temperature/gridded/%%_FV02_%s_%%gridded%%' % deployment_code)
 
     if len(previous_fv02_url) == 1:
         previous_fv02_url = previous_fv02_url[0]
     else:
         previous_fv02_url = ''
 
-    fv01_dir = download_list_nc(list_fv01_url)
+    logger.info("Downloading files:\n%s" % "\n".join(map(str, [os.path.basename(fv01_url) for fv01_url in list_fv01_url])))
+    fv01_dir = download_list_urls(list_fv01_url)
     if incoming_file_path != '':
         # add incoming_file_path from user input arg to list of FV01 files to
         # process
         shutil.copy(incoming_file_path, fv01_dir)
```

Comment:
Good point, I understand that:
> What if the file at incoming_file_path (i.e. exactly same file name) was already downloaded from list_fv01_url? Will this overwrite it?

Yes, it will be overwritten.

> Or what if list_fv01_url contains an older version of the same file?

We're in trouble... I suggest we shouldn't allow this possibility, that is to say that if `incoming_file_path` is not found in `list_fv01_url` then we abort the processing.

I imagine that in a pipeline context we could run the script with an `incoming_file_path` non null in argument after we have sent this same file to S3 and that older version of the file has been discarded.

How does that sound?

Reply:
Actually, the only use of this incoming file is to retrieve a deployment_code from which we can run the script, so maybe best to ignore this file altogether after we got its deployment_code and move on to a regular use of the script when only the deployment_code is given.